internal/terraform/context_apply.go

```diff
@@ -25,6 +25,10 @@ func (c *Context) Apply(plan *plans.Plan, config *configs.Config) (*states.State
 
 	log.Printf("[DEBUG] Building and walking apply graph for %s plan", plan.UIMode)
 
+	// FIXME: refresh plans still store outputs as changes, so we can't use
+	// Empty()
+	possibleRefresh := len(plan.Changes.Resources) == 0
```

Comment:
Is it worth considering using `plan.UIMode == plans.RefreshOnlyMode` as the condition here instead of looking at resource changes? The `UIMode` field does have a "do not use for apply-time logic" warning, so that's rude, but if the goal is detecting a refresh-only plan for this temporary fix maybe that's okay to ignore.

I can't point definitively to an example of how things could go wrong, but my concern is that there might somehow be a non-refresh plan which causes changes to checks without having any resource changes, and we'd be incorrectly recording the checks in state in that case.